main.py

```diff
@@ -1,80 +1,58 @@
 import base64
-import httplib2
-import oauth2
 import random
 from xml.etree import ElementTree as etree
-from urllib.parse import urlencode
 import json
 import os
+import requests
+from requests_oauthlib import OAuth1
+import time
 
 CONSUMER_KEY = os.getenv("CONSUMER_KEY")
 CONSUMER_SECRET = os.getenv("CONSUMER_SECRET")
 ACCESS_KEY = os.getenv("ACCESS_KEY")
 ACCESS_SECRET = os.getenv("ACCESS_SECRET")
+FLICKR_API_KEY = os.getenv("FLICKR_API_KEY")
 
 
-def flickr_query_to_get_list_of_pictures(request_h):
-    """
-    function to get list of photos
-    :param request_h: http stream
-    :return: list of bytestrings for each photo
-    """
+def flickr_get_photos_list():
     payload = {'method': 'flickr.people.getPublicPhotos',
-               'api_key': '1a19ecf3ddf46b3323bca01dc8afbd81',
+               'api_key': FLICKR_API_KEY,
                'user_id': '76163487@N05'
                }
-    resp, content = request_h.request('https://api.flickr.com/services/rest/?&{}'.format(urlencode(payload)), "POST")
-    xml = etree.fromstring(content.decode('utf-8'))
-    xml = xml[0]  # get child of root (list of <photo> tags)
-    photos = [{'id': tag.attrib['id'], 'secret': tag.attrib['secret'],
-               "farm": tag.attrib["farm"], "server": tag.attrib["server"]} for tag in xml]
-    return photos
+    photos_list_response = requests.post('https://api.flickr.com/services/rest', data=payload)
+    return [tag.attrib for tag in etree.fromstring(photos_list_response.text)[0]]
 
 
-def flickr_query2(request_h2, photo_index):
-    """
-    function to load photo from server
-    :rtype: object
-    """
-    resp2, content2 = request_h2.request('https://farm{}.staticflickr.com/{}/{}_{}.jpg'.format(
-        photo_index["farm"], photo_index["server"], photo_index['id'], photo_index["secret"]))
-    return content2
+def flickr_get_photo(photo_attributes):
+    flickr_binary_file = requests.get('https://farm{}.staticflickr.com/{}/{}_{}.jpg'.format(
+        photo_attributes["farm"], photo_attributes["server"], photo_attributes['id'], photo_attributes["secret"]))
+    return {"source": flickr_binary_file.content, "name": photo_attributes["title"]}
 
 
-def twitter_query(media_id):
-    # return 1
-    consumer = oauth2.Consumer(key=CONSUMER_KEY, secret=CONSUMER_SECRET)
-    access_token = oauth2.Token(key=ACCESS_KEY, secret=ACCESS_SECRET)
-    client = oauth2.Client(consumer, access_token)
-    endpoint = "https://api.twitter.com/1.1/statuses/update.json"
-    params = urlencode({'status': 'Hello from Python + Heroku',
-                        'media_ids': media_id})
-    r, c = client.request(endpoint, "POST", params)
-    print(c)
+def upload_picture_to_twitter(pic):
+    endpoint = "https://upload.twitter.com/1.1/media/upload.json"
```

Comment:
Move all endpoints to constants at the beginning of the script. Constants are just variables with ALL_UPPER_CASE_NAME.
For example:
MEDIA_UPLOAD_ENDPOINT = "https://upload.twitter.com/1.1/media/upload.json"

Reply:
I don't see why it will be better. These strings used only once in functions that declaring them so I don't need them in global scope. Also in functions I could instantly see to which endpoint will be send request, instead of scrolling to top of the file.